pkg/alertmanager/multitenant.go

```diff
@@ -243,65 +246,70 @@ func (am *MultitenantAlertmanager) Stop() {
 	am.alertmanagersMtx.Unlock()
 	err := am.peer.Leave(am.cfg.PeerTimeout)
 	if err != nil {
-		level.Warn(util.Logger).Log("msg", "MultitenantAlertmanager: failed to leave the cluster", "err", err)
+		level.Warn(am.logger).Log("msg", "failed to leave the cluster", "err", err)
 	}
-	level.Debug(util.Logger).Log("msg", "MultitenantAlertmanager stopped")
+	level.Debug(am.logger).Log("msg", "stopping")
 }
 
-// Load the full set of configurations from the server, retrying with backoff
+// Load the full set of configurations from the alert store, retrying with backoff
 // until we can get them.
-func (am *MultitenantAlertmanager) loadAllConfigs() map[string]configs.View {
+func (am *MultitenantAlertmanager) loadAllConfigs() map[string]alerts.AlertConfigDesc {
 	backoff := util.NewBackoff(context.Background(), backoffConfig)
 	for {
 		cfgs, err := am.poll()
 		if err == nil {
-			level.Debug(util.Logger).Log("msg", "MultitenantAlertmanager: initial configuration load", "num_configs", len(cfgs))
+			level.Debug(am.logger).Log("msg", "initial configuration load", "num_configs", len(cfgs))
 			return cfgs
 		}
-		level.Warn(util.Logger).Log("msg", "MultitenantAlertmanager: error fetching all configurations, backing off", "err", err)
+		level.Warn(am.logger).Log("msg", "error fetching all configurations, backing off", "err", err)
 		backoff.Wait()
 	}
 }
 
-func (am *MultitenantAlertmanager) updateConfigs(now time.Time) error {
+func (am *MultitenantAlertmanager) updateConfigs() error {
 	cfgs, err := am.poll()
 	if err != nil {
 		return err
 	}
-	am.addNewConfigs(cfgs)
+	am.syncConfigs(cfgs)
 	return nil
 }
 
-// poll the configuration server. Not re-entrant.
-func (am *MultitenantAlertmanager) poll() (map[string]configs.View, error) {
-	configID := am.latestConfig
-	cfgs, err := am.configsAPI.GetAlerts(context.Background(), configID)
+// poll the alert store. Not re-entrant.
+func (am *MultitenantAlertmanager) poll() (map[string]alerts.AlertConfigDesc, error) {
+	cfgs, err := am.store.ListAlertConfigs(context.Background())
 	if err != nil {
-		level.Warn(util.Logger).Log("msg", "MultitenantAlertmanager: configs server poll failed", "err", err)
+		level.Warn(am.logger).Log("msg", "configs alert store poll failed", "err", err)
```

Comment:
I'm not sure this log is useful, because I checked the code and it looks we already load whenever `poll()` returns error, so we're going to end up with double logging.

Reply:
I agree, I'll remove this line.